java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -79,13 +82,21 @@
   private final CallBack callBack;
   private int typeBufferAllocationSizeInBytes;
 
+  private final FieldType fieldType;
+
   private static final byte TYPE_WIDTH = 1;
   private static final FieldType INTERNAL_STRUCT_TYPE = new FieldType(false /*nullable*/,
       ArrowType.Struct.INSTANCE, null /*dictionary*/, null /*metadata*/);
 
+  @Deprecated
   public UnionVector(String name, BufferAllocator allocator, CallBack callBack) {
```

Comment:
Fixed and VarWidth Vectors seem to accept the field and return the same instead of creating one every time.

Is there a reason we do this differently..

Reply:
I think the mainly difference is typeIds[], if we add child vector to union vector after its creation, typeIds[] should be changed when getField.